modules/ting_search_carousel/ting_search_carousel.admin.inc

```diff
@@ -84,116 +84,104 @@ function ting_search_carousel_settings_admin_form($form, &$form_state) {
  *   Form structure.
  */
 function ting_search_carousel_admin_form(array $form, array &$form_state) {
-  $searches_num = !empty($form_state['values']['num_searches']) ? $form_state['values']['num_searches'] : 0;
-  $clicked_btn = '';
-  $remove_btn = '';
-  $hidden = variable_get('ting_carousel_search_queries_hidden', array());
-
-  if (isset($form_state['clicked_button'])) {
-    $clicked_btn = $form_state['clicked_button']['#name'];
-
-    // Get a clicked 'remove' button.
-    foreach ($form_state['buttons'] as $k => $v) {
-      if ($v['#name'] == $clicked_btn) {
-        if (preg_match('/edit-remove/', $v['#id'])) {
-          $remove_btn = $k;
-          $hidden[] = $remove_btn;
-          variable_set('ting_carousel_search_queries_hidden', $hidden);
-          break;
-        }
-      }
-    }
-  }
+  $form['#tree'] = TRUE;
 
   $form['ting_search_carousel'] = array(
     '#type' => 'fieldset',
-    '#title' => 'Searches',
+    '#title' => t('Searches'),
   );
 
-  // Get current saved queries.
-  $searches = variable_get('ting_carousel_search_queries', array());
-
-  $index = 0;
-  $count = 1;
-
-  // Display saved queries.
-  if (isset($searches[0]['title'])) {
-    $count = count($searches);
-    for (; $index < $count; $index++) {
-      $form['ting_search_carousel']['ting_searches']['search_' . $index] = ting_search_carousel_query_form($searches[$index], $index);
-    }
+  if (empty($form_state['num_searches'])) {
+    $form_state['existing_searches'] = variable_get('ting_search_carousel_searches', array());
   }
 
-  // Whether add a new query field.
-  if ($clicked_btn == 'add_query') {
-    $searches_num++;
-  }
+  $form['ting_search_carousel']['ting_searches'] = array(
```

Comment:
Call this `$form['ting_search_carousel_searches']` and move the `#tree => TRUE` and wrapper markup from the form here. That'll make `system_settings_form_submit()` save directly to the variable.